extensions/amp-analytics/0.1/test/test-analytics-root.js

```diff
@@ -333,25 +333,67 @@ describes.realWin('AmpdocAnalyticsRoot', {amp: 1}, (env) => {
       addTestInstance(root3.getElement(body, '#target'), null);
     });
 
-    it('should find an AMP element for AMP search', () => {
+    it('should find an AMP element for AMP search', async () => {
       child.classList.add('i-amphtml-element');
-      return root.getAmpElement(body, '#child').then((element) => {
-        expect(element).to.equal(child);
-      });
+      const element = await root.getAmpElement(body, '#child');
+      expect(element).to.equal(child);
     });
 
-    it('should allow not-found element for AMP search', () => {
-      return root.getAmpElement(body, '#unknown').catch((error) => {
+    it('should allow not-found element for AMP search', async () => {
+      await root.getAmpElement(body, '#unknown').catch((error) => {
         expect(error).to.match(/Element "#unknown" not found/);
       });
     });
 
-    it('should fail if the found element is not AMP for AMP search', () => {
+    it('should fail if the found element is not AMP for AMP search', async () => {
       child.classList.remove('i-amphtml-element');
-      return root.getAmpElement(body, '#child').catch((error) => {
+      await root.getAmpElement(body, '#child').catch((error) => {
         expect(error).to.match(/required to be an AMP element/);
       });
     });
+
+    describe('get amp elements', () => {
+      let child2;
+      let elements;
+
+      beforeEach(() => {
+        child2 = win.document.createElement('child');
+        body.appendChild(child2);
+        child.classList.add('i-amphtml-element');
+        child2.classList.add('i-amphtml-element');
+      });
+
+      it('should find all elements by selector', async () => {
+        child.id = 'myId';
+        child2.id = 'myId';
+        child.classList.add('myClass');
+        child2.classList.add('myClass');
+        elements = await root.getAmpElements(
+          body,
+          ['#myId.myClass'],
+          null,
+          true
+        );
+        expect(elements).to.contain(child);
+        expect(elements).to.contain(child2);
+        expect(elements.length).to.equal(2);
+      });
+
+      it('should allow not-found element for AMP search', async () => {
+        expectAsyncConsoleError(/Element "#unknown" not found/, 1);
```

Comment:
Could you please expand the test a bit, so it covers the case when one of the selector is invalid?